language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/CodeActionUtil.java

```diff
@@ -539,6 +539,12 @@ public static Node largestExpressionNode(Node node, Range range) {
                     return Optional.of(member);
                 }
                 return Optional.empty();
+            } else if (member.kind() == SyntaxKind.RECORD_TYPE_DESC && isWithinBody) {
+                return Optional.of(member);
+            } else if (member.kind() == SyntaxKind.OBJECT_TYPE_DESC && isWithinStartSegment) {
+                return Optional.of(member);
+            } else if (member.kind() == SyntaxKind.METHOD_DECLARATION && isWithinStartSegment) {
+                return Optional.of(member);
```

Comment:
Is it possible to handle all these in a single check?

Reply:
Not really. it changes `isWithinBody` and `isWithinStartSegment` only common case is for OBJECT_TYPE_DESC and METHOD_DECLARATION. But that isn't worth since it expands the if check into multiple lines

Reply:
I was thinking like something as a combination of all three conditions. Let's keep this as it is for the moment then

Reply:
Yeah that we can do. But isn't this more readable? anyway let's address it in a separate PR